src/polymer/expression-scanner.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import * as espree from 'espree';
+import * as estree from 'estree';
+import * as parse5 from 'parse5';
+
+import {HtmlVisitor, ParsedHtmlDocument} from '../html/html-document';
+import {HtmlScanner} from '../html/html-scanner';
+import {baseParseOptions} from '../javascript/javascript-parser';
+import {ScannedFeature, SourceRange} from '../model/model';
+import {Warning} from '../warning/warning';
+
+const p = dom5.predicates;
+const isTemplate = p.hasTagName('template');
+const matchInnerDataBindingTemplate = p.AND(
```

Comment:
Add comment. I thought at first this was going to include templates in `dom-module`s.

Also, any of these can be used standalone, not sure you need the separate dom-bind predicate.